feat: add privacy pattern page

nkrantz opened the pull request:
Add the Privacy Pattern page to the docs site based on [this ticket](https://issues.corp.twilio.com/browse/DSYS-2996).

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

nkrantz commented on packages/paste-website/src/pages/patterns/privacy/index.mdx:
```diff
@@ -0,0 +1,286 @@
+      </HelpText>
+    </Box>
+  </Do>
+  {/* omitted the anchor to GH Discussions in the body here, is that ok? */}
```
There was an anchor in the spec on this "Don't" body text, let me know if there's a way to do that in the body attribute. (I already tried creating a separate `dontBody` template literal in another folder and importing that in to use as the `body` value but that didn't work)

zahnster replied:
Looking at the DoDon't component, the body prop is just a string, doesn't look like there's a way to support html content currently.

katieporter commented on packages/paste-website/src/pages/patterns/privacy/index.mdx:
```diff
@@ -0,0 +1,286 @@
+
+In the API, Twilio clearly labels fields as “PII” and “Not PII”, defines what PII is, and outlines how Twilio handles fields marked "Not PII".
+
+In Console, we currently do not label fields that are not intended to store personal data. This data is then available and exposed in our internal tools, such as Monkey. We need to adequately warn or discourage customers from putting personal data into freeform fields that are not intended for PII. In particular, we’ve seen users input personal data, such as their first and last name, into the Fr ...
```
```suggestion
In our UIs, we currently do not label fields that are not intended to store personal data. This data is then available and exposed in our internal tools, such as Monkey. We need to adequately warn or discourage customers from putting personal data into freeform fields that are not intended for PII. In particular, we’ve seen users input personal data, such as their first and last name, into the Friendly Name field even though it isn't PII.
```

katieporter commented:
This is looking great! Just a couple of tiny things I noticed:
- I didn't see the Privacy pattern in the nav or the Patterns overview page.
- Could we put an example of the pattern at the very top of the page, like we have for the [status pattern](https://paste.twilio.design/patterns/status)?
- The Privacy pattern has gone through peer review and design assets are available, which I've just updated in Airtable - no action needed there, just an FYI.
- I'm not loving how the ingredients section is looking...though I don't love how it looks on most of our patterns pages haha. I'm gonna run this one by the rest of the design team and see if we might just want to switch it to an unordered list or something.

nkrantz commented:
@katieporter addressed your comments! The page will show up in the side nav once we check "has documentation" in the Airtable and the site builds again, I just wanted to wait to check it off until I know that this PR will be the next one to merge (so that it doesn't show up before the page is available in prod).

zahnster reviewed (approved):
👍🏼
